travertino/src/travertino/properties/aliased.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from warnings import filterwarnings, warn
+
+# Make sure deprecation warnings are shown by default
+filterwarnings("default", category=DeprecationWarning)
+
+
+class Condition:
+    def __init__(self, **properties):
+        """A condition describing the values of one or more properties of a style.
+
+        :param properties: Any number of name: value pairs
+        """
+        self.properties = properties
+
+    def match(self, style):
+        return all(style[name] == value for name, value in self.properties.items())
+
+    def __str__(self):
+        return "; ".join(
+            [f"{name} == {value}" for name, value in self.properties.items()]
+        )
+
+
+class aliased_property:
+    def __init__(self, source: str | dict, deprecated: bool = False):
+        """Create a property that aliases an existing property.
+
+        :param source: If this is a string, it is the name of the property to
+            reference. Otherwise, it is a dictionary mapping conditions to the correct
+            property name to use. If no condition is met, an AttributeError is raised.
+        :deprecated: Is this property name deprecated?
+        """
+        self.source = source
+        self.deprecated = deprecated
+
+    def __set_name__(self, owner, name):
+        self.name = name
+        owner._BASE_ALL_PROPERTIES[owner].add(name)
+
+    def __get__(self, obj, objtype=None):
+        if obj is None:
+            return self
+
+        return obj[self.derive_name(obj)]
+
+    def __set__(self, obj, value):
+        if value is self:
+            # This happens during autogenerated dataclass __init__ when no value is
+            # supplied.
+            return
+
+        obj[self.derive_name(obj)] = value
+
+    def __delete__(self, obj):
+        del obj[self.derive_name(obj)]
+
+    def is_set_on(self, obj):
+        return self.derive_name(obj) in obj
+
+    def derive_name(self, obj):
+        name = None
+
+        if isinstance(self.source, str):
+            name = self.source
+        else:
+            for condition, result in self.source.items():
+                if condition.match(obj):
+                    name = result
+                    break
+
+        if name is None:
+            conditions = " or ".join([f"({condition})" for condition in self.source])
```

Comment:
Again - minor comprehension simplification:
```suggestion
            conditions = " or ".join(f"({condition})" for condition in self.source)
```